database/migrations/default/1731003209538_rename_is_coa_roadway/up.sql

```diff
@@ -0,0 +1,151 @@
+                    location_group = 1 -- not level 5
+                    and st_contains(geometry, new.position)
+                limit 1);
+        end if;
+
+        raise debug 'found location: % compared to previous location: %', new.location_id, old.location_id;
+        --
+        -- check if in austin full purpose jurisdiction
+        --
+        new.in_austin_full_purpose =  st_contains((select geometry from geo.atd_jurisdictions where id = 5), new.position);
+        raise debug 'in austin full purpose: % compared to previous: %', new.in_austin_full_purpose, old.in_austin_full_purpose;
+        --
+        -- get council district
+        --
+        new.council_district = (
+            select
+                council_district
+            from
+                geo.council_districts
+            where
+                st_contains(geometry, new.position)
+            limit 1);
+        raise debug 'council_district: % compared to previous: %', new.council_district, old.council_district;
+        --
+        -- get engineering area
+        --
+        new.engineering_area_id = (
+            select
+                engineering_area_id
+            from
+                geo.engineering_areas
+            where
+                st_contains(geometry, new.position)
+            limit 1);
+        raise debug 'engineering_area_id: % compared to previous: %', new.engineering_area_id, old.engineering_area_id;
+        --
+        -- get signal engineer area
+        --
+        new.signal_engineer_area_id = (
+            select
+                signal_engineer_area_id
+            from
+                geo.signal_engineer_areas
+            where
+                st_contains(geometry, new.position)
+            limit 1);
+        raise debug 'signal_engineer_area_id: % compared to previous: %', new.signal_engineer_area_id, old.signal_engineer_area_id;
+        --
+        -- get zip code
+        --
+        new.zipcode = (
+            select
+                zipcode
+            from
+                geo.zip_codes
+            where
+                st_contains(geometry, new.position)
+            limit 1);
+        raise debug 'zipcode: % compared to previous: %', new.zipcode, old.zipcode;
+        --
+        -- get apd sector
+        --
+        new.apd_sector_id = (
+            select
+                primary_key
+            from
+                geo.apd_sectors
+            where
+                st_contains(geometry, new.position)
+            limit 1);
+        raise debug 'apd_sector_id: % compared to previous: %', new.apd_sector_id, old.apd_sector_id;
+        --
+        -- check if is_coa_roadway
+        --
+        if ((new.in_austin_full_purpose or new.rpt_city_id = 22) and not new.private_dr_fl) then
+            new.is_coa_roadway = not st_contains((select geometry from geo.non_coa_roadways), new.position);
+            raise debug 'is_coa_roadway: % compared to previous: %', new.is_coa_roadway, old.is_coa_roadway;
```

Comment:
here is the updated business logic

Reply:
@johnclary I might be testing this incorrectly, but I'm getting hung up on what is supposed to happen when toggling the private drive flag from `false` to `true` since the code after `then` won't run when `new.private_dr_fl` is `true` here. Should this run in either case and recalculate `is_coa_roadway`?

Could be that my Friday brain is struggling. 😅

Reply:
great catch, Mike—thank you! i just pushed two commits:
- add the missing `else` case to this block of code
- add `rpt_city_id` to the conditions that cause this trigger function to run, so that editing the crash's city actually has an affect

Reply:
awesome - i am now seeing the trigger fire in step 5. 🚀